### PR TITLE
Fixed issue #18358: Statistics filter feature is not working properly

### PR DESCRIPTION
**Pull request by gabrieljenik**
For encrypted attributes, filtergin requires encrypting values for the query.
Only did for single choice and multiple choice.

### CODE REVIEW

File: application/helpers/admin/statistics_helper.php [hunk 1]
```diff
@@ -380,7 +382,7 @@ function buildSelects($allfields, $surveyid, $language)
                 $thisquestion = Yii::app()->db->quoteColumnName($pv) . " IN (";
 
                 foreach ($_POST[$pv] as $condition) {
-                    $thisquestion .= "'$condition', ";
+                    $thisquestion .= "'" . getEncryptedCondition($responseModel, $pv, $condition) . "', ";
```
Comment (Shnoulle) [inline application/helpers/admin/statistics_helper.php:385]: usage of https://www.yiiframework.com/doc/api/1.1/CDbConnection#quoteValue-detail seems better, no ?

File: application/helpers/admin/statistics_helper.php [hunk 2]
```diff
@@ -401,7 +403,8 @@ function buildSelects($allfields, $surveyid, $language)
                 foreach ($aresult as $arow) {
                     // only add condition if answer has been chosen
                     if (in_array($arow['title'], $_POST[$pv])) {
-                        $mselects[] = Yii::app()->db->quoteColumnName(substr($pv, 1, strlen($pv)) . $arow['title']) . " = 'Y'";
+                        $fieldname = substr($pv, 1, strlen($pv)) . $arow['title'];
+                        $mselects[] = Yii::app()->db->quoteColumnName($fieldname) . " = '" . getEncryptedCondition($responseModel, $fieldname, 'Y') . "'";
```
Comment (Shnoulle) [inline application/helpers/admin/statistics_helper.php:407]: usage of https://www.yiiframework.com/doc/api/1.1/CDbConnection#quoteValue-detail seems better, no ?

**Review by Shnoulle — CHANGES_REQUESTED** [review]
usage of https://www.yiiframework.com/doc/api/1.1/CDbConnection#quoteValue-detail seems better, no ?

Why only single choice since issue is same for contains etc …(where usage of quoteValue is needed too)

**Comment by Shnoulle** [comment]
> > Why only single choice since issue is same for contains etc …(where usage of quoteValue is needed too)
> 
> We didn't fix the quoteValue issue (want minimal impact), just handled the encryption for the single choice.

And it's not possible for contains or upper/lower thah : i'm silly …

Just the usage of quoteValue https://www.yiiframework.com/doc/api/1.1/CDbConnection#quoteValue-detail ?

File: application/helpers/admin/statistics_helper.php (hunk 2 above)
Comment (c-schmitz) [reply application/helpers/admin/statistics_helper.php:407]: yeah, quoteValue. I agree.

**Comment by Shnoulle** [comment]
> We didn't fix the quoteValue issue (want minimal impact), just handled the encryption for the single choice.

There are a difference between answer value : a lot of restriction are done : can contain only [A-Za-z0-9]

What for encrypted value ? Can not contain strange carater ?
(and quoteValue seems really a good option : always)

**Comment by Shnoulle** [comment]
PS : merge with last master to have the CI Pipeline green too.

**Review by Shnoulle — APPROVED** [review]

File: application/helpers/admin/statistics_helper.php [hunk 3]
```diff
@@ -402,7 +404,9 @@ function buildSelects($allfields, $surveyid, $language)
                 foreach ($aresult as $arow) {
                     // only add condition if answer has been chosen
                     if (in_array($arow['title'], $_POST[$pv])) {
-                        $mselects[] = Yii::app()->db->quoteColumnName(substr($pv, 1, strlen($pv)) . $arow['title']) . " = 'Y'";
+                        $fieldname = substr($pv, 1, strlen($pv)) . $arow['title'];
+                        $condition =
```
Comment (c-schmitz) [inline application/helpers/admin/statistics_helper.php:408]: What is going on here?

**Review by c-schmitz — CHANGES_REQUESTED** [review]

**Comment by gabrieljenik** [comment]
Updated.

**Pull request merged**